src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExerciseTestCase.java

```diff
@@ -231,4 +233,29 @@ public String toString() {
         return "ProgrammingExerciseTestCase{" + "id=" + getId() + ", testName='" + testName + '\'' + ", weight=" + weight + ", active=" + active + ", visibility=" + visibility
                 + ", bonusMultiplier=" + bonusMultiplier + ", bonusPoints=" + bonusPoints + '}';
     }
+
+    /**
+     * Check if the provided test was found in the result's feedbacks with positive = true.
+     *
+     * @param result of the build run.
+     * @return true if there is a positive feedback for a given test.
+     */
+    public boolean isSuccessful(Result result) {
```

Comment:
Can you check if Jackson adds this the the JSON-response? If yes, we should probably ignore it. same below.
```suggestion
    @JsonIgnore
    public boolean isSuccessful(Result result) {
```